caf-audit/src/test/java/com/hpe/caf/auditing/elastic/ElasticAuditIT.java

```diff
@@ -328,38 +328,94 @@ public void testMultiTenantESIndexing() throws Exception
+            indices[i] = indices[i].toLowerCase();
+        }
 
-                tenant2IndexHits = searchDocumentInIndex(transportClient, tenant2IndexId,
-                        APP_ID_FIELD.concat(KEYWORD_SUFFIX), testApplicationId);
+        RetryElasticsearchOperation retryDelete = new RetryElasticsearchOperation();
+        while (retryDelete.shouldRetry()) {
+            try {
+                boolean didElasticAckDelete = client.admin().indices().delete(
+                        new DeleteIndexRequest(indices)).get().isAcknowledged();
 
-                //  Expecting a single hit.
-                Assert.assertTrue(tenant2IndexHits.length == 1);
+                if (didElasticAckDelete) {
+                    // If Elastic acknowledged our delete wait a second to allow it time to delete the indices
+                    Thread.sleep(1000);
+                    break;
+                }
 
-                //  Delete test indexes after verification is complete.
-                deleteTenantIndex(transportClient, tenant1IndexId);
-                deleteTenantIndex(transportClient, tenant2IndexId);
+                // Retry deletion if Elastic did not acknowledge the delete request.
+                try {
+                    retryDelete.retryNeeded();
+                } catch (Exception e) {
+                    throw new RuntimeException(e);
+                }
+
+            } catch (InterruptedException e) {
+                throw new RuntimeException(e);
+            } catch (ExecutionException e) {
+                throw new RuntimeException(e);
             }
         }
     }
 
+    private static SearchHit[] searchDocumentInIndices(TransportClient client, String[] indices, String field,
+                                                       String value)
+    {
+        // Lowercase each string in the array of indices
+        for (int i = 0; i < indices.length; i++) {
+            indices[i] = indices[i].toLowerCase();
+        }
+
+        RetryElasticsearchOperation retrySearch = new RetryElasticsearchOperation();
+        SearchHit[] hits = null;
+        while (retrySearch.shouldRetry()) {
+            hits = client.prepareSearch(indices)
+                    .setTypes(ES_TYPE)
+                    .setSearchType(SearchType.QUERY_THEN_FETCH)
+                    .setQuery(QueryBuilders.matchQuery(field, value.toLowerCase()))
+                    .setFrom(0).setSize(10)
+                    .execute()
+                    .actionGet().getHits().getHits();
+
+            if (hits.length > 0) {
+                break;
+            }
+
+            //  No search hits just yet. Retry.
+            try {
+                retrySearch.retryNeeded();
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+        }
+
+        return hits;
+    }
+
     private static SearchHit[] searchDocumentInIndex(TransportClient client, String index, String field, String value)
```

Comment:
Drop this method, i.e. searchDocumentInIndex and simply re-use searchDocument (as discussed above).